src/register_map.rs

```diff
@@ -0,0 +1,38 @@
+use std::vec::Vec;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use bytecode::instructions::Reg;
+
+pub struct RegisterMap {
+    registers: RefCell<Vec<Reg>>,
```

Comment:
Is there a reason for the `RefCell` here? Is it because you don't want the methods below to me `&mut self`?

Reply:
Yes. This was needed in order to do [this](https://github.com/rbartlensky/Lua-interpreter/pull/1/files/c55dd9d55bc34dd9c82804a3770b52b520f2a41a..86e1ef05ddc0b59c68308b963bf4029fb0bc90d3#diff-f6d8127e28ccb889d76a83aa953d1389R32)

Reply:
Hmm, why does getting a register make a new one? I have a feeling we have made this more complex than we probably wanted it to be. In the next PR, let's try and concentrate on cleaning things up if we can.